prow/pod-utils/gcs/upload.go

```diff
@@ -190,6 +198,17 @@ func DataUploadWithOptions(newReader ReaderFunc, attrs pkgio.WriterOptions) Uplo
 			e = utilerrors.NewAggregate(errors)
 		}()
 
+		if writer.compressData() {
```

Comment:
Why should we prefer this out here vs closed over inside of `writer.ApplyWriterOptions()`?

Reply:
No particular reason. I thought about putting it there, but opted to keep that method simple. I can move it there if it is preferred.